python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -487,7 +487,8 @@ class RandomForestClassifier(Base):
 
     def _predict_model_on_gpu(self, X, output_class,
                               threshold, algo,
-                              num_classes, convert_dtype):
+                              num_classes, convert_dtype,
+                              sparse_forest):
```

Comment:
Could you rename it to `storage_type`, with the same parameter values as in FIL? In this case, no extra handling is required.

Reply:
Actually I requested the name change. `storage_type` is not very descriptive for the Python API and the capital letter strings are not pythonic at all

Reply:
The additional handling is minimal and worth it for a more Pythonic API

Reply:
However, in this case, the parameter is inconsistent between FIL and RF, which is also not idea. @JohnZed What is your opinion on this?

@danteg Lowercase versions of values are also supported in FIL's `storage_type`, e.g. `storage_type='auto'`.

Reply:
Thinking of it, what about naming it `forest_type`, with the same values as for FIL's `storage_type`? Maybe it makes sense to rename it to `forest_type` in FIL as well...

Reply:
yeah that was @vishalmehta1991 suggestion, I think that could be fine. I like the possibility of a semi boolean True/False/'auto', but your suggestion works very well as well. Also making it the same in both places would be a good idea. Wasn't aware that it could take lowercase version of the variables here, the (python) docstring didn't make that obvious or clear.

Reply:
Regarding FIL renaming: I think it's best to do in a separate pull request, to avoid breaking changes. Maybe we need to introduce the new parameter and deprecate the old one first, and to remove it completely later.

Reply:
Totally agreed on a new PR and doing the deprecation

Reply:
I like `fil_sparse_format` (with current true/false/auto) or `fil_storage_type`. I think it's helpful to indicate that these parameters relate to FIL storage. I would vote against using `forest_type` as a name here or in FIL since it's such a general term. If somebody saw `forest_type="REGRESSION"` it would look reasonable, but I don't think others are equally open to interpretation.

Reply:
Updated the variable name to `fil_sparse_format` will create a new PR to update the variable name in FIL as well